### PR TITLE
[Mac] Remove Network option from Preferences

### PR DESCRIPTION
**Pull request by mrward**
Use an environment variable instead of a options page in Preferences
to allow Mono's HttpClientHandler to be used instead of Xamarin.Mac's
NSUrlSessionHandler. Since this is an advanced setting providing
a way in the UI to change this is not required.

Fixes VSTS #796914 - Remove Network option from Preferences

### CODE REVIEW

File: main/src/addins/MacPlatform/MacPlatformSettings.cs [hunk 1]
```diff
@@ -23,14 +23,17 @@
...
-using MonoDevelop.Core;
+using System;
 
 namespace MonoDevelop.MacIntegration
 {
 	static class MacPlatformSettings
 	{
-		public static readonly ConfigurationProperty<bool> UseNSUrlSessionHandler
-			= ConfigurationProperty.Create ("MonoDevelop.MacIntegration.UseNSUrlSessionHandler", true);
+		static MacPlatformSettings ()
```
Comment (Therzok) [inline main/src/addins/MacPlatform/MacPlatformSettings.cs]: Do we need a static constructor? Can't we just inline the value?

Comment (mrward) [reply main/src/addins/MacPlatform/MacPlatformSettings.cs]: Updated.

**Comment by mrward** [comment]
Tried re-running the tests several times but the MacPlatform Tests seem to be timing out. Seem unrelated to this change. Also the tests are sometimes timing out on master.

**Comment by slluis** [comment]
@monojenkins backport release-8.0-integration

**Pull request merged**